Assets/MRTK/Providers/LeapMotion/Editor/ConfigurationChecker/LeapMotionConfigurationChecker.cs

```diff
@@ -0,0 +1,263 @@
+            {
+                // Names of all Leap Motion editor directories that need assembly definitions
+                // Key: Asmdef name
+                // Value: References for the asmdef
+                Dictionary<string, string[]> leapEditorDirectories = new Dictionary<string, string[]>
+                {
+                    { "LeapMotion.Core.Editor", new string[] { "LeapMotion" } },
+                    { "LeapMotion.Core.Scripts.Animation.Editor", new string[] { "LeapMotion", "LeapMotion.Core.Editor", "LeapMotion.Core.Scripts.Utils.Editor" } },
+                    { "LeapMotion.Core.Scripts.Attachments.Editor", new string[] { "LeapMotion", "LeapMotion.Core.Editor" } },
+                    { "LeapMotion.Core.Scripts.Attributes.Editor", new string[] { "LeapMotion" } },
+                    { "LeapMotion.Core.Scripts.DataStructures.Editor", new string[] { "LeapMotion" } },
+                    { "LeapMotion.Core.Scripts.EditorTools.Editor", new string[] { "LeapMotion", "LeapMotion.Core.Scripts.Utils.Editor" } },
+                    { "LeapMotion.Core.Scripts.Utils.Editor", new string[] { "LeapMotion", "LeapMotion.Core.Editor" } },
+                    { "LeapMotion.Core.Scripts.XR.Editor", new string[] { "LeapMotion", "LeapMotion.Core.Editor" } }
+                };
+
+                string pathDifference = GetPathDifference();
+
+                foreach (KeyValuePair<string, string[]> leapAsmDef in leapEditorDirectories)
+                {
+                    // Convert asmdef name to a path
+                    string leapAsmDefPath = leapAsmDef.Key.Replace('.', '/');
+
+                    string leapAsmDefFilename = string.Concat(leapAsmDef.Key, ".asmdef");
+
+                    string fullLeapAsmDefPath = Path.Combine(Application.dataPath, pathDifference, leapAsmDefPath, leapAsmDefFilename);
+
+                    if (!File.Exists(fullLeapAsmDefPath))
+                    {
+                        // Create and save the new asmdef
+                        AssemblyDefinition leapEditorAsmDef = new AssemblyDefinition
+                        {
+                            Name = leapAsmDef.Key,
+                            References = leapAsmDef.Value,
+                            IncludePlatforms = new string[] { "Editor" }
+                        };
+
+                        leapEditorAsmDef.Save(fullLeapAsmDefPath);
+                    }
+                }
+
+                AssetDatabase.Refresh();
+            }
+        }
+
+        [MenuItem("Mixed Reality Toolkit/Utilities/Update/Configure CSC File for Leap Motion", false, 0)]
+        static void UpdateCSC()
+        {
+            string fileName = Path.Combine(Application.dataPath, "csc.rsp");
+
+            FileInfo file = new FileInfo(fileName);
+
+            bool readOnly = (file.Exists) ? file.IsReadOnly : false;
+            if (readOnly)
+            {
+                file.IsReadOnly = false;
```

Comment:
Were you observing a case where the file was marked as read only? I'm curious if we actually just should bail out or log a warning if that happens - generally speaking if things are marked as read only I think we should honor the filesystem state, and not try to mutate things from under the developer

Reply:
This section was following a pattern in the AssemblyDefinition file, I observed that the asmdef files are read only and need to be modified but the csc.rsp is has not been marked as read only.